packages/fx-core/src/plugins/resource/bot/plugin.ts

```diff
@@ -220,15 +224,21 @@ export class TeamsBotImpl {
     // 1. Do bot registration.
     if (this.config.scaffold.wayToRegisterBot === WayToRegisterBot.CreateNew) {
       await handler?.next(ProgressBarConstants.PROVISION_STEP_BOT_REG);
-      await this.createNewBotRegistrationOnAzure();
+      const botAuthCreds = await this.createOrGetBotAppRegistration();
+
+      if (!isArmSupportEnabled()) {
+        await this.provisionBotServiceOnAzure(botAuthCreds);
+      }
     }
 
-    await handler?.next(ProgressBarConstants.PROVISION_STEP_WEB_APP);
-    // 2. Provision azure web app for hosting bot project.
-    await this.provisionWebApp();
+    if (!isArmSupportEnabled()) {
+      await handler?.next(ProgressBarConstants.PROVISION_STEP_WEB_APP);
```

Comment:
Can this if 234 combined with L229's if

Reply:
No. You can see the previous if-clause is inside the `if (this.config.scaffold.wayToRegisterBot === WayToRegisterBot.CreateNew)`